internal/services/machinelearning/machine_learning_workspace_network_outbound_rule_service_tag_resource_test.go

```diff
@@ -0,0 +1,245 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type WorkspaceNetworkOutboundRuleServiceTagResource struct{}
+
+func TestAccMachineLearningWorkspaceNetworkOutboundRuleServiceTag_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_machine_learning_workspace_network_outbound_rule_service_tag", "test")
+	r := WorkspaceNetworkOutboundRuleServiceTagResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccMachineLearningWorkspaceNetworkOutboundRuleServiceTag_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_machine_learning_workspace_network_outbound_rule_service_tag", "test")
+	r := WorkspaceNetworkOutboundRuleServiceTagResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("service_tag").HasValue("AppService"),
+				check.That(data.ResourceName).Key("protocol").HasValue("UDP"),
+				check.That(data.ResourceName).Key("port_ranges").HasValue("445"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccMachineLearningWorkspaceNetworkOutboundRuleServiceTag_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_machine_learning_workspace_network_outbound_rule_service_tag", "test")
+	r := WorkspaceNetworkOutboundRuleServiceTagResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("service_tag").Exists(),
+				check.That(data.ResourceName).Key("protocol").Exists(),
+				check.That(data.ResourceName).Key("port_ranges").Exists(),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r WorkspaceNetworkOutboundRuleServiceTagResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := managednetwork.ParseOutboundRuleID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.MachineLearning.ManagedNetwork.SettingsRuleGet(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
```

Comment:
We don't check for 404 when checking for the existence of the resource in the acceptance tests.
```suggestion
```